src/components/buttons/z-button/index.tsx

```diff
@@ -26,42 +32,45 @@ export class ZButton {
   @Prop({ reflect: true }) disabled?: boolean = false;
   /** HTML button type attribute. */
   @Prop() type?: HTMLButtonElement["type"] = ButtonTypeEnum.button;
-  /** Graphical variant: `primary`, `secondary`, `tertiary`, `dark-bg`. Defaults to `primary`. */
-  @Prop({ reflect: true }) variant?: ButtonVariantBean = ButtonVariantEnum.primary;
+  /** Graphical variant: `primary`, `secondary`, `tertiary`. Defaults to `primary`. */
+  @Prop({ reflect: true }) variant?: ButtonVariantBean =
+    ButtonVariantEnum.primary;
   /** `z-icon` name to use (optional). */
   @Prop() icon?: string;
   /** Available sizes: `big`, `small` and `x-small`. Defaults to `big`. */
   @Prop({ reflect: true })
   size?: ButtonSizeEnum = ButtonSizeEnum.big;
-  /** Reduce button size (deprecated).
-   * @deprecated Use `size` prop.
-   */
-  @Prop({ reflect: true }) issmall?: boolean = false;
-  /** Spy to render square button. */
+  /** set label text  */
   @Prop({ reflect: true })
-  square?: boolean = false;
+  text?: string;
 
   render() {
-    this.hostElement.style.pointerEvents = this.disabled ? "none" : "auto";
-
-    return (
-      <slot name="element">
-        <button
+    if (this.href)
+      return (
+        <a
+          aria-label={this.ariaLabel}
+          href={this.href}
+          target={this.target}
           id={this.htmlid}
-          name={this.name}
-          type={this.type}
-          disabled={this.disabled}
-          class={classNames(
-            this.variant,
-            this.size,
-            { issmall: this.issmall },
-            { square: this.square }
-          )}
+          class={classNames(this.variant, this.size)}
         >
           {this.icon && <z-icon name={this.icon} width={16} height={16} />}
-          <slot />
-        </button>
-      </slot>
+          {this.text && <span>{this.text}</span>}
+        </a>
+      );
+
+    return (
+      <button
+        aria-label={this.ariaLabel}
+        id={this.htmlid}
+        name={this.name}
+        type={this.type}
+        disabled={this.disabled}
+        class={classNames(this.variant, this.size)}
+      >
+        {this.icon && <z-icon name={this.icon} width={16} height={16} />}
+        {this.text && <span>{this.text}</span>}
```

Comment:
Queste due righe mi sembrano uguali sia per `button` che per `a`, magari si possono mettere in un metodo da riutilizzare nei due punti.

Reply:
Per questo, secondo me, avrebbe senso se fosse una parte di codice lunga, cosi da ridurre il numero di righe totali. Essendo solo 4 non è conveniente ma questo è il mio parere.  Ti ho convinto? 😄

Reply:
va bene, non è una cosa fondamentale, se di solito segui questa logica lascia pure così